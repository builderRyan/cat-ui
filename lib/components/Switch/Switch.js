"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) if (e.indexOf(p[i]) < 0)
            t[p[i]] = s[p[i]];
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var classnames_1 = require("classnames");
var react_1 = require("react");
var lodash_1 = require("lodash");
var styled_1 = require("./styled");
/**
 * Switch
 */
var Switch = /** @class */ (function (_super) {
    __extends(Switch, _super);
    function Switch(props) {
        var _this = _super.call(this, props) || this;
        _this.toggle = function () {
            var onClick = _this.props.onClick;
            var checked = !_this.state.checked;
            _this.setChecked(checked);
            onClick(checked);
        };
        _this.handleKeyDown = function (e) {
            if (e.keyCode === 37) { // Left
                _this.setChecked(false);
            }
            else if (e.keyCode === 39) { // Right
                _this.setChecked(true);
            }
        };
        // Handle auto focus when click switch in Chrome
        _this.handleMouseUp = function (e) {
            if (_this.node) {
                _this.node.blur();
            }
            if (_this.props.onMouseUp) {
                _this.props.onMouseUp(e);
            }
        };
        _this.focus = function () {
            _this.node.focus();
        };
        _this.blur = function () {
            _this.node.blur();
        };
        _this.saveNode = function (node) {
            _this.node = node;
        };
        var checked = false;
        if ('checked' in props) {
            checked = !!props.checked;
        }
        else {
            checked = !!props.defaultChecked;
        }
        _this.state = { checked: checked };
        _this.switchRef = React.createRef();
        return _this;
    }
    Switch.prototype.componentDidMount = function () {
        var _a = this.props, autoFocus = _a.autoFocus, disabled = _a.disabled;
        this.node = this.switchRef.current;
        if (autoFocus && !disabled) {
            this.focus();
        }
    };
    Switch.getDerivedStateFromProps = function (nextProps) {
        if ('checked' in nextProps) {
            return { checked: !!nextProps.checked };
        }
        return null;
    };
    Switch.prototype.setChecked = function (checked) {
        if (this.props.disabled) {
            return;
        }
        if (!('checked' in this.props)) {
            this.setState({ checked: checked });
        }
        this.props.onChange(checked);
    };
    Switch.prototype.render = function () {
        var _a = this.props, className = _a.className, disabled = _a.disabled, loadingIcon = _a.loadingIcon, checkedChildren = _a.checkedChildren, unCheckedChildren = _a.unCheckedChildren, restProps = __rest(_a, ["className", "disabled", "loadingIcon", "checkedChildren", "unCheckedChildren"]);
        var checked = this.state.checked;
        var calsses = classnames_1.default('hmly-switch', className, {
            'hmly-switch-checked': checked,
            'hmly-switch-disabled': disabled
        });
        return (React.createElement(styled_1.StyledSwitch, __assign({}, restProps, { type: 'button', role: 'switch', "aria-checked": checked, disabled: disabled, className: calsses, ref: this.saveNode, onKeyDown: this.handleKeyDown, onClick: this.toggle, onMouseUp: this.handleMouseUp }),
            loadingIcon,
            React.createElement("span", { className: 'hmly-switch-inner' }, checked ? checkedChildren : unCheckedChildren)));
    };
    Switch.defaultProps = {
        // checkedChildren: '',
        // unCheckedChildren: '',
        className: '',
        defaultChecked: false,
        onChange: lodash_1.noop,
        onClick: lodash_1.noop
    };
    return Switch;
}(react_1.Component));
exports.default = Switch;
