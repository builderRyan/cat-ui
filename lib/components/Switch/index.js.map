{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../src/components/Switch/index.ts"],"names":[],"mappings":";;AAAA,mCAA6B;AAE7B,kBAAe,gBAAM,CAAA"}