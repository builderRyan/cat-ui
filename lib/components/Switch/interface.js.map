{"version":3,"file":"interface.js","sourceRoot":"","sources":["../../../src/components/Switch/interface.tsx"],"names":[],"mappings":""}