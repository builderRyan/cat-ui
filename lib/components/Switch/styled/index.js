"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
Object.defineProperty(exports, "__esModule", { value: true });
var styled_components_1 = require("styled-components");
var duration = '.2s';
exports.rcSwitchOn = styled_components_1.keyframes(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  0% {\n    transform: scale(1);\n  }\n  50% {\n    transform: scale(1.25);\n  }\n  100% {\n    transform: scale(1.1);\n  }\n"], ["\n  0% {\n    transform: scale(1);\n  }\n  50% {\n    transform: scale(1.25);\n  }\n  100% {\n    transform: scale(1.1);\n  }\n"])));
exports.rcSwitchOff = styled_components_1.keyframes(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n  0% {\n    transform: scale(1.1);\n  }\n  100% {\n    transform: scale(1);\n  }\n"], ["\n  0% {\n    transform: scale(1.1);\n  }\n  100% {\n    transform: scale(1);\n  }\n"])));
var StyledSwitch = styled_components_1.default.button(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n  position: relative;\n  display: inline-block;\n  box-sizing: border-box;\n  width: 36px;\n  height: 22px;\n  line-height: 20px;\n  padding: 0;\n  vertical-align: middle;\n  border-radius: 20px 20px;\n  border: 1px solid #d8d8d8;\n  background-color: #d8d8d8;\n  cursor: pointer;\n  transition: all ", " cubic-bezier(0.35, 0, 0.25, 1);\n\n  .hmly-switch-inner {\n    color:#fff;\n    font-size: 12px;\n    position: absolute;\n    left: 24px;\n    top: 0;\n  }\n\n  &:after {\n    position: absolute;\n    width: 18px;\n    height: 18px;\n    left: 2px;\n    top: 1px;\n    border-radius: 50% 50%;\n    background-color: #fff;\n    content: \" \";\n    cursor: pointer;\n    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.26);\n    transform: scale(1);\n    transition: left ", " cubic-bezier(0.35, 0, 0.25, 1);\n    animation-timing-function: cubic-bezier(0.35, 0, 0.25, 1);\n    animation-duration: ", ";\n    animation-name: ", ";\n  }\n\n  &:hover:after {\n    transform: scale(1.1);\n    animation-name: ", ";\n  }\n\n  &:focus {\n    box-shadow: 0 0 0 2px tint(#2db7f5, 80%);\n    outline: none;\n  }\n\n  &.hmly-switch-checked {\n    border: 1px solid #333;\n    background-color: #333;\n\n    .hmly-switch-inner {\n      left: 6px;\n    }\n\n    &:after{\n      left: 15px;\n    }\n  }\n\n  &.hmly-switch-disabled {\n    cursor: no-drop;\n    background: #ccc;\n    border-color: #ccc;\n\n    &:after{\n      background: #9c9c9c;\n      animation-name: none;\n      cursor: no-drop;\n    }\n\n    &:hover:after{\n      transform: scale(1);\n      animation-name: none;\n    }\n  }\n\n  .hmly-switch-label {\n    display: inline-block;\n    line-height: 20px;\n    font-size: 14px;\n    padding-left: 10px;\n    vertical-align: middle;\n    white-space: normal;\n    pointer-events: none;\n    user-select: text;\n  }\n"], ["\n  position: relative;\n  display: inline-block;\n  box-sizing: border-box;\n  width: 36px;\n  height: 22px;\n  line-height: 20px;\n  padding: 0;\n  vertical-align: middle;\n  border-radius: 20px 20px;\n  border: 1px solid #d8d8d8;\n  background-color: #d8d8d8;\n  cursor: pointer;\n  transition: all ", " cubic-bezier(0.35, 0, 0.25, 1);\n\n  .hmly-switch-inner {\n    color:#fff;\n    font-size: 12px;\n    position: absolute;\n    left: 24px;\n    top: 0;\n  }\n\n  &:after {\n    position: absolute;\n    width: 18px;\n    height: 18px;\n    left: 2px;\n    top: 1px;\n    border-radius: 50% 50%;\n    background-color: #fff;\n    content: \" \";\n    cursor: pointer;\n    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.26);\n    transform: scale(1);\n    transition: left ", " cubic-bezier(0.35, 0, 0.25, 1);\n    animation-timing-function: cubic-bezier(0.35, 0, 0.25, 1);\n    animation-duration: ", ";\n    animation-name: ", ";\n  }\n\n  &:hover:after {\n    transform: scale(1.1);\n    animation-name: ", ";\n  }\n\n  &:focus {\n    box-shadow: 0 0 0 2px tint(#2db7f5, 80%);\n    outline: none;\n  }\n\n  &.hmly-switch-checked {\n    border: 1px solid #333;\n    background-color: #333;\n\n    .hmly-switch-inner {\n      left: 6px;\n    }\n\n    &:after{\n      left: 15px;\n    }\n  }\n\n  &.hmly-switch-disabled {\n    cursor: no-drop;\n    background: #ccc;\n    border-color: #ccc;\n\n    &:after{\n      background: #9c9c9c;\n      animation-name: none;\n      cursor: no-drop;\n    }\n\n    &:hover:after{\n      transform: scale(1);\n      animation-name: none;\n    }\n  }\n\n  .hmly-switch-label {\n    display: inline-block;\n    line-height: 20px;\n    font-size: 14px;\n    padding-left: 10px;\n    vertical-align: middle;\n    white-space: normal;\n    pointer-events: none;\n    user-select: text;\n  }\n"])), duration, duration, duration, exports.rcSwitchOff, exports.rcSwitchOn);
exports.StyledSwitch = StyledSwitch;
var templateObject_1, templateObject_2, templateObject_3;
//# sourceMappingURL=index.js.map