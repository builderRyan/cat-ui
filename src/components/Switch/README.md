# Switch 开关

## Example

```jsx
<Switch checked={checked} onChange={handleChange} /> 
```

## API

|       属性        |             类型             | 默认值  |      描述      | 是否必填 |
| :---------------: | :--------------------------: | :-----: | :------------: | :------: |
|      preifx       |           `string`           |  `cat`  |      前缀      |    -     |
|     className     |           `string`           |    -    |      类名      |    -     |
|       style       |    `React.CSSProperties`     |    -    |    行内样式    |    -     |
|     disabled      |          `boolean`           | `false` |    是否禁用    |    -     |
|  checkedChildren  |      `React.ReactNode`       |    -    |  选中时的内容  |    -     |
| unCheckedChildren |      `React.ReactNode`       |    -    | 非选中时的内容 |    -     |
|  defaultChecked   |          `boolean`           |    -    |  默认是否选中  |    -     |
|      checked      |          `boolean`           |    -    |    是否选中    |    -     |
|     autoFocus     |          `boolean`           | `false` |    是否聚焦    |    -     |
|     onChange      | `(checked: boolean) => void` |    -    |  变化时的回调  |    -     |
