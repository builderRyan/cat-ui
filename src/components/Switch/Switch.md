Example

```jsx
class Example extends React.Component {
  constructor (props) {
    super(props)
    this.state = {}
    this.onSwitchChange = this.onSwitchChange.bind(this)
  }

  onSwitchChange (value) {
    console.log(123)
    this.setState({
      switchOpen: value,
      earlyAccessDisplayTime: value ? 86400 : 0
    })
  }

  render () {
    const { switchOpen } = this.state
    console.log(switchOpen)
    return (<div className='switch-box'>
      <Switch onChange={this.onSwitchChange } checked={switchOpen} />
      <style>
        {`
          .switch-box {
            padding: 5px;
          }
        `}
      </style>
    </div>
    )
  }
}

<Example />
```
