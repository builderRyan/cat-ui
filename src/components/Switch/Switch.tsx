import React from 'react';
import classNames from 'classnames';
import { Component } from 'react';
import noop from 'lodash/noop';
import PropTypes from 'prop-types';
import omit from 'lodash/omit';

export interface SwitchProps extends Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, 'onChange'> {
  /**
   * 类名前缀
   */
  prefix: string;
  /** 
   * 类名
   */
  className?: string;
  /** 
   * 样式
   */
  style?: React.CSSProperties;
  /** 
   * 禁用
   */
  disabled?: boolean;
  /** 
   * 选中时的内容
   */
  checkedChildren?: React.ReactNode;
  /** 
   * 非选中时的内容
   */
  unCheckedChildren?: React.ReactNode;
  /** 
   * 变化时的回调
   */
  onChange?: (checked: boolean) => void;
  /** 
   * 是否激活
   */
  checked?: boolean;
  /** 
   * 默认值
   */
  defaultChecked?: boolean;
  /** 
   * 自动聚焦
   */
  autoFocus: boolean;
}

export interface SwitchState {
  checked: boolean;
}

/**
 * Switch
 */
class Switch extends Component<SwitchProps, SwitchState> {
  static propTypes = {
    prefix: PropTypes.string,
    className: PropTypes.string,
    style: PropTypes.object,
    disabled: PropTypes.bool,
    checkedChildren: PropTypes.node,
    unCheckedChildren: PropTypes.node,
    onChange: PropTypes.func,
    checked: PropTypes.bool,
    defaultChecked: PropTypes.bool,
    autoFocus: PropTypes.bool,
  }

  static defaultProps = {
    prefix: 'cat',
    disabled: false,
    onChange: noop,
    defaultChecked: false,
    autoFocus: false,
  };

  private node: HTMLButtonElement;

  constructor(props: SwitchProps) {
    super(props);
    let checked = false;
    if ('checked' in props) {
      checked = !!props.checked;
    } else {
      checked = !!props.defaultChecked;
    }
    this.state = {
      checked
    };
  }

  componentDidMount() {
    const { autoFocus, disabled } = this.props;
    if (autoFocus && !disabled) {
      this.focus();
    }
  }

  static getDerivedStateFromProps(nextProps: SwitchProps) {
    if ('checked' in nextProps) {
      return { checked: !!nextProps.checked };
    }
    return null;
  }

  setChecked(checked: boolean) {
    if (this.props.disabled) {
      return;
    }
    if (!('checked' in this.props)) {
      this.setState({ checked });
    }

    const { onChange } = this.props;

    onChange && onChange(checked);
  }

  handleClick = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    const { onClick } = this.props;
    const checked = !this.state.checked;
    this.setChecked(checked);
    onClick && onClick(event);
  };

  handleKeyDown = (e: React.KeyboardEvent<HTMLButtonElement>) => {
    if (e.keyCode === 37) {
      // Left
      this.setChecked(false);
    } else if (e.keyCode === 39) {
      // Right
      this.setChecked(true);
    }
  };

  // Handle auto focus when click switch in Chrome
  handleMouseUp = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    if (this.node) {
      this.blur();
    }
    if (this.props.onMouseUp) {
      this.props.onMouseUp(e);
    }
  };

  focus = () => {
    this.node.focus();
  };

  blur = () => {
    this.node.blur();
  };

  getButtonNode = (node: HTMLButtonElement) => {
    this.node = node;
  };

  render() {
    const {
      prefix,
      className,
      disabled,
      checkedChildren,
      unCheckedChildren,
      ...restProps
    } = this.props;
    const checked = this.state.checked;

    const classPrefix = `${prefix}-switch`;
    const calsses = classNames(
      classPrefix,
      {
        'cat-switch--checked': checked,
        'cat-switch--disabled': disabled
      },
      className
    );

    const buttonProps = omit(restProps, ['onChange'])

    return (
      <button
        {...buttonProps}
        type="button"
        role="switch"
        aria-checked={checked}
        disabled={disabled}
        className={calsses}
        ref={this.getButtonNode}
        onKeyDown={this.handleKeyDown}
        onClick={this.handleClick}
        onMouseUp={this.handleMouseUp}
      >
        <span className="cat-switch__inner">
          {checked ? checkedChildren : unCheckedChildren}
        </span>
      </button>
    );
  }
}

export default Switch;
