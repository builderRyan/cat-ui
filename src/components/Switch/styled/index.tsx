import styled,  { keyframes }  from 'styled-components'
import { primaryBtnColor } from '../../../styles/theme'

const duration: string = '.2s'

export const rcSwitchOn = keyframes`
  0% {
    transform: scale(1);
  }
  50% {
    transform: scale(1.25);
  }
  100% {
    transform: scale(1.1);
  }
`;

export const rcSwitchOff = keyframes`
  0% {
    transform: scale(1.1);
  }
  100% {
    transform: scale(1);
  }
`;


const StyledSwitch = styled.button`
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  width: 36px;
  height: 22px;
  line-height: 20px;
  padding: 0;
  vertical-align: middle;
  border-radius: 20px 20px;
  border: 1px solid #d8d8d8;
  background-color: #d8d8d8;
  cursor: pointer;
  transition: all ${duration} cubic-bezier(0.35, 0, 0.25, 1);

  .hmly-switch-inner {
    color:#fff;
    font-size: 12px;
    position: absolute;
    left: 24px;
    top: 0;
  }

  &:after {
    position: absolute;
    width: 18px;
    height: 18px;
    left: 2px;
    top: 1px;
    border-radius: 50% 50%;
    background-color: #fff;
    content: " ";
    cursor: pointer;
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.26);
    transform: scale(1);
    transition: left ${duration} cubic-bezier(0.35, 0, 0.25, 1);
    animation-timing-function: cubic-bezier(0.35, 0, 0.25, 1);
    animation-duration: ${duration};
    animation-name: ${rcSwitchOff};
  }

  &:hover:after {
    transform: scale(1.1);
    animation-name: ${rcSwitchOn};
  }

  &:focus {
    box-shadow: 0 0 0 2px tint(#2db7f5, 80%);
    outline: none;
  }

  &.hmly-switch-checked {
    border: 1px solid #333;
    background-color: #333;

    .hmly-switch-inner {
      left: 6px;
    }

    &:after{
      left: 15px;
    }
  }

  &.hmly-switch-disabled {
    cursor: no-drop;
    background: #ccc;
    border-color: #ccc;

    &:after{
      background: #9c9c9c;
      animation-name: none;
      cursor: no-drop;
    }

    &:hover:after{
      transform: scale(1);
      animation-name: none;
    }
  }

  .hmly-switch-label {
    display: inline-block;
    line-height: 20px;
    font-size: 14px;
    padding-left: 10px;
    vertical-align: middle;
    white-space: normal;
    pointer-events: none;
    user-select: text;
  }
`;

export { StyledSwitch }
  