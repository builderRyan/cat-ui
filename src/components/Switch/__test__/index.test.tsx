import React from 'react';
import Enzyme from 'enzyme';
import { mount } from 'enzyme';
import Switch from '../index';
import sinon from 'sinon';


describe('Switch', () => {
  // reset dom
  beforeEach(() => {
    document.body.innerHTML = '';
  });
  afterEach(() => {
    document.body.innerHTML = '';
  });

  test('renders correctly', () => {
    const wrapper = mount(
      <Switch />
    )
    expect(wrapper).toMatchSnapshot()
  });

  test('defaultChecked and checked props', () => {
    const wrapper1 = mount(
      <Switch defaultChecked />
    );
    expect(wrapper1.state('checked')).toBe(true);

    const wrapper2 = mount(
      <Switch checked />
    );
    expect(wrapper2.state('checked')).toBe(true);

    const wrapper3 = mount(
      <Switch defaultChecked={false} checked={true} />
    );
    expect(wrapper3.state('checked')).toBe(true);
  });

  test('disabled props', () => {
    const wrapper = mount(
      <Switch disabled={true} />
    );
    expect(wrapper.state('checked')).toBe(false);

    const button = wrapper.find('button');
    button.simulate('click');

    expect(wrapper.state('checked')).toBe(false);
  });

  test('auto focus', () => {
    const wrapper = mount(
      <Switch autoFocus></Switch>
    );

    const buttonDOM = wrapper.find('button').getDOMNode();
    const activeElement = document.activeElement;

    expect(buttonDOM).toBe(activeElement)
  });

  test('When mouseup is triggered, the switch loses focus', () => {
    const wrapper = mount(
      <Switch></Switch>
    );

    const button = wrapper.find('button');
    button.simulate('click');

    const activeElement = document.activeElement;

    expect(button).not.toBe(activeElement)
  });

  test('The left and right keys switch the Switch state correctly', () => {
    const wrapper = mount(
      <Switch></Switch>
    );

    expect(wrapper.state('checked')).toBe(false);
    const button = wrapper.find('button');
    // right
    button.simulate('keydown', {
      keyCode: 39,
    });
    expect(wrapper.state('checked')).toBe(true);
    // left
    button.simulate('keydown', {
      keyCode: 37,
    });
    expect(wrapper.state('checked')).toBe(false);
  });

  test('callbacks triggered', () => {
    const mouseupCallback = sinon.spy();

    const wrapper = mount(
      <Switch onMouseUp={mouseupCallback}></Switch>
    );

    expect(mouseupCallback.callCount).toBe(0);

    const button = wrapper.find('button');
    button.simulate('mouseup');

    expect(mouseupCallback.callCount).toBe(1);



  });
})
