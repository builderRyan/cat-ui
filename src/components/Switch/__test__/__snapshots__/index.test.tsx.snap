// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders correctly 1`] = `
<Switch
  autoFocus={false}
  defaultChecked={false}
  disabled={false}
  onChange={[Function]}
  prefix="cat"
>
  <button
    aria-checked={false}
    autoFocus={false}
    className="cat-switch"
    defaultChecked={false}
    disabled={false}
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseUp={[Function]}
    role="switch"
    type="button"
  >
    <span
      className="cat-switch__inner"
    />
  </button>
</Switch>
`;
